sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchemaSuite.scala

```diff
@@ -192,4 +192,13 @@ class CSVInferSchemaSuite extends SparkFunSuite with SQLHelper {
     Seq("en-US").foreach(checkDecimalInfer(_, StringType))
     Seq("ko-KR", "ru-RU", "de-DE").foreach(checkDecimalInfer(_, DecimalType(7, 0)))
   }
+
+  test("SPARK-34953 - DateType should be inferred when user defined format are provided") {
+    val options = new CSVOptions(Map("dateFormat" -> "dd-MM-yyyy",
```

Comment:
does it work with default date pattern?

Reply:
Yes this works with default date patter which is 'yyyy-MM-dd', If we don't specify in the option dateFormat option it will check with the default value of dateFormat.

`2,23232,hello,2016-11-11`

```
val p =  spark.read.format("csv").option("header", "false").option("delimiter", ",").option("inferSchema", "true").load("/testDir/test.csv")

p.schema
res0: org.apache.spark.sql.types.StructType = StructType(StructField(_c0,IntegerType,true), StructField(_c1,IntegerType,true), StructField(_c2,StringType,true), StructField(_c3,DateType,true))
```